src/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -438,75 +447,95 @@ public Task AuthenticateAsServerAsync(SslServerAuthenticationOptions sslServerAu
             }
         }
 
-        public override bool IsEncrypted
-        {
-            get
-            {
-                return IsAuthenticated;
-            }
-        }
+        public override bool IsEncrypted => IsAuthenticated;
 
-        public override bool IsSigned
-        {
-            get
-            {
-                return IsAuthenticated;
-            }
-        }
+        public override bool IsSigned => IsAuthenticated;
 
-        public override bool IsServer
-        {
-            get
-            {
-                return _sslState.IsServer;
-            }
-        }
+        public override bool IsServer => _context != null && _context.IsServer;
 
         public virtual SslProtocols SslProtocol
         {
             get
             {
-                return _sslState.SslProtocol;
-            }
-        }
+                CheckThrow(true);
+                SslConnectionInfo info = _context.ConnectionInfo;
+                if (info == null)
+                {
+                    return SslProtocols.None;
+                }
 
-        public virtual bool CheckCertRevocationStatus
-        {
-            get
-            {
-                return _sslState.CheckCertRevocationStatus;
+                SslProtocols proto = (SslProtocols)info.Protocol;
+                SslProtocols ret = SslProtocols.None;
+
+#pragma warning disable 0618 // Ssl2, Ssl3 are deprecated.
+                // Restore client/server bits so the result maps exactly on published constants.
+                if ((proto & SslProtocols.Ssl2) != 0)
+                {
+                    ret |= SslProtocols.Ssl2;
+                }
+
+                if ((proto & SslProtocols.Ssl3) != 0)
+                {
+                    ret |= SslProtocols.Ssl3;
+                }
+#pragma warning restore
+
+                if ((proto & SslProtocols.Tls) != 0)
+                {
+                    ret |= SslProtocols.Tls;
+                }
+
+                if ((proto & SslProtocols.Tls11) != 0)
+                {
+                    ret |= SslProtocols.Tls11;
+                }
+
+                if ((proto & SslProtocols.Tls12) != 0)
+                {
+                    ret |= SslProtocols.Tls12;
+                }
+
+                if ((proto & SslProtocols.Tls13) != 0)
+                {
+                    ret |= SslProtocols.Tls13;
+                }
+
+                return ret;
             }
         }
 
+        public virtual bool CheckCertRevocationStatus => _context != null && _context.CheckCertRevocationStatus != X509RevocationMode.NoCheck;
+
+        //
+        // This will return selected local cert for both client/server streams
+        //
         public virtual X509Certificate LocalCertificate
         {
             get
             {
-                return _sslState.LocalCertificate;
+                CheckThrow(true);
+                return InternalLocalCertificate;
```

Comment:
I realize you're just keeping this as it was, but this doesn't seem like it needs to be in its own function and could just be manually inlined here as happens for a bunch of the other properties.  This looks to be the only call site.